src/repr/src/row.rs

```diff
@@ -2853,36 +2859,33 @@ impl SharedRow {
         I: IntoIterator<Item = D>,
         D: Borrow<Datum<'a>>,
     {
-        let binding = Self::SHARED_ROW.with(Rc::clone);
-        let mut row_builder = binding.borrow_mut();
+        let mut row_builder = Self::get();
         let mut row_packer = row_builder.packer();
         row_packer.extend(iter);
         row_builder.clone()
     }
-
-    /// Calls the provided closure with a [`RowPacker`] writing the shared row.
-    ///
-    /// # Panics
-    ///
-    /// Panics when the row is already borrowed elsewhere.
-    pub fn pack_with<F, R>(&mut self, f: F) -> R
-    where
-        for<'a> F: FnOnce(&'a mut RowPacker<'a>) -> R,
-    {
-        let mut borrow = self.borrow_mut();
-        let mut packer = borrow.packer();
-        (f)(&mut packer)
-    }
 }
 
 impl std::ops::Deref for SharedRow {
-    type Target = RefCell<Row>;
+    type Target = Row;
 
     fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
 
+impl std::ops::DerefMut for SharedRow {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl Drop for SharedRow {
+    fn drop(&mut self) {
+        Self::SHARED_ROW.set(Some(std::mem::take(&mut self.0)))
```

Comment:
doesn't set also call drop on the inner value of the `Cell`? so wouldn't we be calling drop on `Option<SharedRow>` which would also be `self`?

better question, is it worth explicitly testing this behaviour if possible?

Reply:
`SHARED_ROW.set(..)` does drop the value that is currently contained in the cell, but that value is always `None` when this drop function runs, so nothing is actually dropped. This is because the existence of an instance of `SharedRow` must have gotten through `SharedRow::get`, the only constructor, and that constructor *moves* the shared row contained in the thread local into the returned instance and leaves a `None` value in the cell.

What kind of test did you have in mind?

Reply:
ahhhhhhhhh! ty! can we add that as a comment? I guess we could for a test:
```rust
let shared_row = SharedRow::get();
/// insert some data into shared_row
drop(shared_row);
assert!(SharedRow::get().is_empty());
```

also might be nice to test that we panic with a legible message when if we try to double borrow.